jwst/tweakreg/tweakreg_step.py

```diff
@@ -170,70 +165,61 @@ def process(self, input):
         else:
             align_to_abs_refcat = False
 
-        if len(images) == 0:
-            raise ValueError("Input must contain at least one image model.")
-
-        # Build the catalogs for input images
-        for image_model in images:
-            if use_custom_catalogs and image_model.meta.tweakreg_catalog:
+            # since we're not aligning to a reference catalog, check if we
+            # are saving catalogs, if not, and we have 1 group, skip
+            if not self.save_catalogs and n_groups == 1:
+                # we need at least two exposures to perform image alignment
+                self.log.warning("At least two exposures are required for image "
+                                 "alignment.")
+                self.log.warning("Nothing to do. Skipping 'TweakRegStep'...")
+                self.skip = True
+                for model in images:
+                    model.meta.cal_step.tweakreg = "SKIPPED"
+                return input
+
+        # === start processing images ===
+
+        # pre-allocate collectors (same length and order as images)
+        correctors = [None] * len(images)
+
+        # Build the catalog and corrector for each input images
+        for (model_index, image_model) in enumerate(images):
+            # now that the model is open, check it's metadata for a custom catalog
+            # only if it's not listed in the catdict
+            if use_custom_catalogs and image_model.meta.filename not in catdict:
+                if (image_model.meta.tweakreg_catalog is not None and image_model.meta.tweakreg_catalog.strip()):
+                    catdict[image_model.meta.filename] = image_model.meta.tweakreg_catalog
+            if use_custom_catalogs and catdict.get(image_model.meta.filename, None) is not None:
+                # FIXME this modifies the input_model
```

Comment:
This note is a reminder that this part of the first iteration through the models modifies the input models. This modification will either need to be deferred or temporary files written (when using a `ModelContainer` with models that aren't kept in memory). I'm on the fence about keeping this comment. On one hand it serves as a reminder when this step is later updated to be compatible with a "on-disk" `ModelContainer`. On the other hand it's not that informative without more context (which I'm not sure makes sense to add to the comment).

If preferred I can remove this (and the other similar comments).